File: backend/FlashIdBackend/Application/Common/Services/QrFieldDefinitions.cs
namespace Application.Common.Services;

public static class QrFieldDefinitions
{
    public static readonly IReadOnlyList<string> IdentityDocumentMandatoryFields = new List<string>
    {
        "Date of birth", "Photograph",
    };

    public static readonly IReadOnlyList<string> IdentityDocumentOptionalFields = new List<string>
    {
        "Identity number", "Full surname", "Full forenames","Citizenship status","Gender", "Country of birth", "Signature", "Card issue date and number",
    };

    public static readonly IReadOnlyList<string> DriversLicenseMandatoryFields = new List<string>
    {
        "Photo", "Expiry date", "Date of birth",
    };

    public static readonly IReadOnlyList<string> DriversLicenseOptionalFields = new List<string>
    {
        "Full name", "SA ID number", "License number", "License code","Country of issue", "Signature", "Vehicle restrictions", "Date of issue",
    };
}

File: web/src/services/qr-service/qr-field-definitions.ts
export type CredentialType = 'identityDocument' | 'driversLicense'

export const MANDATORY_FIELDS: Record<CredentialType, string[]> = {
  identityDocument: ['Date of birth', 'Photograph'],
  driversLicense: ['Photo', 'Expiry date', 'Date of birth'],
}

export const OPTIONAL_FIELDS: Record<CredentialType, string[]> = {
  identityDocument: [
    'Identity number',
    'Full surname',
    'Full forenames',
    'Citizenship status',
    'Gender',
    'Country of birth',
    'Signature',
    'Card issue date and number',
  ],
  driversLicense: [
    'Full name',
    'SA ID number',
    'License number',
    'License code',
    'Country of issue',
    'Signature',
    'Vehicle restrictions',
    'Date of issue',
  ],
}
